### PR TITLE
Add functionality to update Cluster and AzureCluster

### PR DESCRIPTION
**Pull request by tuommaki**
This change performs required changes to update Cluster and AzureCluster CRs after new supplementary CRs have been created.

### CODE REVIEW

**Review by calvix — APPROVED** [review]

**Review by whites11 — APPROVED** [review]

File: pkg/migration/azure_crs.go [hunk 1]
```diff
@@ -391,6 +396,158 @@ func (m *azureMigrator) readRelease(ctx context.Context, ver string) error {
...
+
+	// Drop operator version label.
+	delete(cluster.Labels, label.AzureOperatorVersion)
+
+	// Drop finalizers.
+	cluster.Finalizers = nil
+
+	// Adjust k8s apiserver bind port to match kubeadm.
+	if cluster.Spec.ClusterNetwork != nil && cluster.Spec.ClusterNetwork.APIServerPort != nil {
+		cluster.Spec.ClusterNetwork.APIServerPort = to.Int32Ptr(6443)
```
Comment (kopiczko) [inline pkg/migration/azure_crs.go:410]: Shouldn't that be set even if `cluster.Spec.ClusterNetwork.APIServerPort` is nil?

File: pkg/migration/azure_crs.go [hunk 2]
```diff
@@ -391,6 +396,158 @@ func (m *azureMigrator) readRelease(ctx context.Context, ver string) error {
...
+	cluster := m.crs.azureCluster
+
+	// Drop operator version label.
+	delete(cluster.Labels, label.AzureOperatorVersion)
+
+	// Drop finalizers.
+	cluster.Finalizers = nil
+
+	// Use default credentials.
+	cluster.Spec.IdentityRef = nil
```
Comment (kopiczko) [inline pkg/migration/azure_crs.go:437]: Can you explain that line?

File: pkg/migration/azure_crs.go [hunk 3]
```diff
@@ -391,6 +396,158 @@ func (m *azureMigrator) readRelease(ctx context.Context, ver string) error {
...
+						Name: fmt.Sprintf("%s-%s-%s-%s", cluster.Name, "API", "PublicLoadBalancer", "PublicIP"),
+					},
+				},
+			},
+		}
+	}
+
+	// START OF UGLY HACK **************************************************
+	// XXX: This is just a shortcut to make testing ergonomics better. Final
+	// code should be rewritten.
```
Comment (kopiczko) [inline pkg/migration/azure_crs.go:457]: Why do we need that?

File: pkg/migration/azure_crs.go [hunk 4]
```diff
@@ -391,6 +396,158 @@ func (m *azureMigrator) readRelease(ctx context.Context, ver string) error {
...
+	if len(cluster.Spec.NetworkSpec.APIServerLB.FrontendIPs) == 0 {
+		ip := capz.FrontendIP{
+			Name: fmt.Sprintf("%s-%s-%s-%s", cluster.Name, "API", "PublicLoadBalancer", "Frontend"),
+			PublicIP: &capz.PublicIPSpec{
+				Name: fmt.Sprintf("%s-%s-%s-%s", cluster.Name, "API", "PublicLoadBalancer", "PublicIP"),
+			},
+		}
+
+		cluster.Spec.NetworkSpec.APIServerLB.FrontendIPs = append(cluster.Spec.NetworkSpec.APIServerLB.FrontendIPs, ip)
```
Comment (kopiczko) [inline pkg/migration/azure_crs.go:466]: Following ifs will cover the rest.

```suggestion
		cluster.Spec.NetworkSpec.APIServerLB.FrontendIPs = append(cluster.Spec.NetworkSpec.APIServerLB.FrontendIPs, capz.FrontendIP{})
```

File: pkg/migration/azure_crs.go (hunk 1 above)
Comment (tuommaki) [reply pkg/migration/azure_crs.go:410]: It's optional. CAPZ defaults to `6443`: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/b5ad5c4c8156cd02eb5447e4405c412c1b9dc64b/azure/scope/cluster.go#L461

File: pkg/migration/azure_crs.go (hunk 2 above)
Comment (tuommaki) [reply pkg/migration/azure_crs.go:437]: Whoops, that's leftover from my earlier testing. We must not nullify that anymore. Earlier when I tested by moving to separate cluster, the credentials were pre-configured there so resetting that simplified the testing setup & migration code when I focused on solving the etcd migration problem. I'll drop that.

File: pkg/migration/azure_crs.go (hunk 3 above)
Comment (tuommaki) [reply pkg/migration/azure_crs.go:457]: You mean rewrite? My original thinking was that we don't do hacky thingie for final migration but that has changed. All good now, I'll just remove the comment.

Comment (kopiczko) [reply pkg/migration/azure_crs.go]: Then you can probably remove while FrontendIP content above

**Review by kopiczko — APPROVED** [review]

**Pull request merged**